packages/@aws-cdk/aws-appsync/lib/runtime.ts

```diff
@@ -0,0 +1,47 @@
+import { CfnFunctionConfiguration } from './appsync.generated';
+
+/**
+ * Appsync supported runtimes. Only JavaScript as of now
+ */
+export enum FunctionRuntimeFamily {
+  /**
+   * AppSync JavaScript runtime
+   */
+  JS = 'APPSYNC_JS',
+}
+
+/**
+ * Utility class for specifying specific appsync runtime versions
+ */
+export class FunctionRuntime {
+  /**
+   * APPSYNC_JS v1.0.0 runtime
+   */
+  public static readonly JS_1_0_0 = new FunctionRuntime(FunctionRuntimeFamily.JS, '1.0.0');
+
+  /**
+   * The name of the runtime
+   */
+  public readonly name: string;
+
+  /**
+   * The runtime version
+   */
+  public readonly version: string;
+
+  public constructor(family: FunctionRuntimeFamily, version: string) {
+    this.name = family;
+    this.version = version;
+  }
+
+  /**
+   * Convert to Cfn runtime configuration property format
+   */
+  public toProperties(): CfnFunctionConfiguration.AppSyncRuntimeProperty {
```

Comment:
I don't think we should expose the L1 layer since this is a public API. I think
we should create a new interface that matches.